torch_geometric/data/hypergraph_data.py

```diff
@@ -0,0 +1,118 @@
+import copy
+import warnings
+from typing import Any, List, Optional
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data
+from torch_geometric.typing import EdgeType, NodeType, OptTensor
+from torch_geometric.utils import hyper_subgraph, select
+
+
+class HyperGraphData(Data):
+    def __init__(self, x: OptTensor = None, edge_index: OptTensor = None,
```

Comment:
Add comments like done in `Data` object [here](https://github.com/pyg-team/pytorch_geometric/blob/a4447449a7ab185a643dba878e8548c452a62a26/torch_geometric/data/data.py#L397). Try to mention the `edge_index` and how it should look for hyper graphs.